src/spaces/grid_multi.jl

```diff
@@ -133,7 +133,9 @@ Base.IteratorSize(::Type{<:GridSpaceIdIterator}) = Base.SizeUnknown()
 # Instructs how to combine two positions. Just to avoid code duplication for periodic
 combine_positions(pos, origin, ::GridSpaceIdIterator{false}) = pos .+ origin
 function combine_positions(pos, origin, iter::GridSpaceIdIterator{true})
-    mod1.(pos .+ origin, iter.space_size)
+    off_pos = pos .+ origin
+    checkbounds(Bool, iter.stored_ids, off_pos...) && return off_pos
```

Comment:
can we add a documentation comment here? Something like "it is faster to check for bounds and only if they fail call the `mod` function. Because, most positions do not need the `mod` function, and the `mod` function is expensive. So overall this leads to better perfroamnce.

Reply:
Yes, that's something would be useful for someone reading the code!